software-mansion/radon-ide
# Add custom device names

This PR introduces following enhancements:
- Adds visual differentiation to the active device in `DevicesSelect` and `ManageDevicesView` by bolding the device name and using a brighter icon
- Allows to change the active device directly from the `ManageDevicesView` 
- Adds an option to assign a `device name` during the creation of a new device using the `CreateDeviceView` (supported for both Android and iOS devices)
- Adds an option to rename a `device name` in the `ManageDevicesView`
- Renames `displayName` to `deviceName` in `DeviceInfo`
- Update text and background colors in disabled fields in the `CreateDeviceView` 

Custom names are limited to alphanumeric chars, spaces, dashes, and underscores and length between 1 and 30 characters.

In Android's config.ini, the "avd.ini.displayname" key is used to store the device name, which can store any string. During emulator creation we pass to 'hw.device.name' key model's identifier which later is used by emulator to display device status bar layout correctly.  The `mapIdToModel` function is used to translate model identifiers into names within the webview.

⚠️ When you need to add a new supported device, make sure to also include the device identifier in the `consts.ts`. Identifiers are sourced from `hw.device.name` in `config.ini` for Android and `deviceType` in `device.plist` for iOS.

Test plan:
1. Switch the active device in the `DevicesSelect` and `ManageDevicesView`.
2. Create new iOS and Android devices with names different than device model names, try to include illegal characters or names longer than max acceptable length.
3. Add a second device with the same name.
4. Rename iOS and Android device' names.
5. Restart the development environment (RNIDE) and verify if all newly added devices are present and accessible.

screenshots:
<img width="527" alt="Screenshot 2024-10-29 at 12 48 37" src="https://github.com/user-attachments/assets/c5580508-0220-40b6-8384-4292f58d128d">
<img width="607" alt="Screenshot 2024-10-15 at 10 58 11" src="https://github.com/user-attachments/assets/f727f283-bbc3-4fed-a6b0-168079c92f9b">
<img width="321" alt="Screenshot 2024-10-28 at 12 33 49" src="https://github.com/user-attachments/assets/11e10293-4811-47dd-a5de-397448ed82cd">
<img width="522" alt="Screenshot 2024-10-28 at 12 35 27" src="https://github.com/user-attachments/assets/681cb791-e68a-49c4-a4c6-1f4788b598f0">
<img width="527" alt="Screenshot 2024-10-29 at 12 49 26" src="https://github.com/user-attachments/assets/d2d7eddd-5345-488d-9200-de6b0263686a">
<img width="527" alt="Screenshot 2024-10-29 at 12 48 51" src="https://github.com/user-attachments/assets/8b78973f-2d6c-4b50-ab72-216f8f8d31a3">
